Solutions/Corvus.Azure.Storage.Tenancy/Corvus/Azure/Storage/Tenancy/TableStorageConfiguration.cs

```diff
@@ -0,0 +1,46 @@
+﻿// <copyright file="TableStorageConfiguration.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+namespace Corvus.Azure.Storage.Tenancy
+{
+    /// <summary>
+    /// Encapsulates configuration for a storage account.
+    /// </summary>
+    public class TableStorageConfiguration
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TableStorageConfiguration"/> class.
+        /// </summary>
+        public TableStorageConfiguration()
+        {
+        }
+
+        /// <summary>
+        /// Gets or sets the account name.
+        /// </summary>
+        /// <remarks>If the account key secret name is empty, then this should contain a complete connection string.</remarks>
+        public string? AccountName { get; set; }
```

Comment:
I presume these are all nullable because a) `Microsoft.Extensions.Configuration` binding doesn't support ctor-based construction, and b) we can't quite be bothered to write the "nullable backing field, non-nullable property" solution.

It's OK to decide that the long hard slog isn't worth it, but if that's what we're doing I prefer it to be explicit: instead of declaring that these are nullable by design, just admit that proper nullable-reference-awareness is too much effort to be worth doing here, and use `#nullable annotations disable` for this section to declare that we're opting out. It's a more honest statement of what's going on than slapping `?` on everything.